graphql.py

```diff
@@ -0,0 +1,92 @@
+
+# This is a simple tutorial to show how graphQL API can be implemented in python
+"""
+Step 1: Create Schema
+GraphQL schema for extracting results from a website.
+"""
+import graphene
+import extraction
+import requests
+
+def extract(url):
+    html = requests.get(url).text
+    extracted = extraction.Extractor().extract(html, source_url=url) # Gets HTLML page attributes
+    # print(extracted)
+    return extracted
+
+class Website(graphene.ObjectType):
+    url = graphene.String(required=True)
+    title = graphene.String()
+    description = graphene.String()
+    image = graphene.String()
+    feed = graphene.String()
+
+    
+class Query(graphene.ObjectType):
+    website = graphene.Field(Website, url=graphene.String())
+
+    def resolve_website(parent, info, url): # resolver for 'website' field in schema
+        extracted = extract(url)
+        return Website(url=url,
+                       title=extracted.title,
+                       description=extracted.description,
+                       image=extracted.image,
+                       feed=extracted.feed
+        )
+      
+schema = graphene.Schema(query=Query)
+
+
+"""
+Step 2: Run GraphQL on Flask
+Python HTTP server for GraphQL.
+"""
+from flask import Flask
+from flask_graphql import GraphQLView
+
+
+app = Flask(__name__)
+app.add_url_rule('/', view_func=GraphQLView.as_view('graphql', schema=schema, graphiql=True))
+app.run()
+
+
+"""
+Step 3: Call the API with a query
+Python HTTP server for GraphQL.
+"""
+import pprint
+
+def query_url(url):
+    q = """
+    {
+      website (url: "https://gmail.com" ) {    
+        url
+        title
+        image
+      }
+    }
```

Comment:
Maybe use triple single quotes for this to separate the comments from strings.

Reply:
Would be a better idea when we build this in production. For a demo, this is fine.